### DISCUSSION TITLE
Broken auth for private go mod on selfhosted gitlab

### DISCUSSION THREAD

**Discussion by thejan2009** [renovatebot/renovate#12552] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Community Edition 14.4.1

### Describe the bug

Probably related to https://github.com/renovatebot/renovate/issues/12339. We used to handle authentication for go mod using a custom .netrc with username `gitlab-ci-token` and the access token. If I understand the code correctly, authentication parameters are now passed to go command and they override .netrc configuration. But the new username is either `token` or `x-access-token` which is not what the gitlab api expects.

I guess this could be fixed by also allowing username and password configuration for hosts instead of just tokens.

### Relevant debug logs

<details><summary>Logs</summary>

```
Command failed: go get -d ./...
go: <gitlab>/<project>@v0.0.0-20210817125733-8bf4add26c77: invalid version: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/77bab64dc8532aac7077d4adc04c7d23b3e06bee94faeb21a0eb08591ddd0981: exit status 128:
	fatal: could not read Password for 'https://**redacted**@<gitlab>': terminal prompts disabled

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Please create a feature request for support for go modules username/password auth, if that's what you think will fix it. Please be sure to specify instructions for how to reproduce and how your config would look, even if you can't "publicly" reproduce.

**Reply by thejan2009** [reply]
That was just an idea, I'm not sure if it's the best solution. Below is a config excerpt that worked until recently. Would not overriding .netrc config also be an option?

```
    - echo "machine <gitlab> login gitlab-ci-token password $RENOVATE_TOKEN" > ~/.netrc
    - >
      renovate
      --platform="gitlab"
      --endpoint="${CI_API_V4_URL}"
      --git-author="${RENOVATE_GIT_AUTHOR}"
      ...
```

**Reply by rarkins** [reply]
No, not if it's incompatible with our environment variable approach as you've indicated

**Comment by thejan2009** [comment, marked as answer]
Fixed by https://github.com/renovatebot/renovate/pull/13216
